### PR TITLE
chore: update token factory

### PR DESCRIPTION
**Pull request by markusahlstrand**
This is basically just moving over the new and better token factory from the token service. It is using a mock for the token generation so we can remove the fixture we had before. 

I commented out the refresh-token part in this PR as I wanted to keep the scope down a bit. Not sure if we want to add it eventually?

### CODE REVIEW

File: src/authentication-flows/silent.ts [hunk 1]
```diff
@@ -35,7 +36,7 @@ export interface SilentAuthParams {
 interface SuperState {
   userId: string;
   authParams: AuthParams;
-  user: User;
+  user: Profile;
```
Comment (markusahlstrand) [inline src/authentication-flows/silent.ts:39]: So.. in the auth service it's a bit different. We have a profile which corresponds more to what the User is in the token service.

File: src/helpers/generate-auth-response.ts [hunk 2]
```diff
@@ -1,26 +1,36 @@
...
-    name?: string;
-    givenName?: string;
-    familyName?: string;
-    nickname?: string;
-  };
-  code_challenge_method?: CodeChallengeMethod;
-  code_challenge?: string;
+}
+
+export interface GenerateAuthResponseParamsForCode
```
Comment (markusahlstrand) [inline src/helpers/generate-auth-response.ts:19]: This is the stuff moved over from the token service.

File: src/helpers/generate-auth-response.ts [hunk 3]
```diff
@@ -42,62 +50,80 @@ export async function generateCode({
...
+      clientId: authParams.client_id,
+      userId: userId,
+      iss: env.ISSUER,
+      sid,
+      nonce: nonce || authParams.nonce,
+    });
+  }
+
+  // REFRESH TOKEN
+  // if (authParams.scope?.split(' ').includes('offline_access')) {
```
Comment (markusahlstrand) [inline src/helpers/generate-auth-response.ts:108]: No refresh token support in auth for now..

File: src/server.ts [hunk 4]
```diff
@@ -28,7 +26,6 @@ const server = {
         sendEmail,
         stateFactory: State.getFactory(env.STATE, env),
         userFactory: User.getFactory(env.USER, env),
-        TokenFactory: createTokenFactory(getToken),
```
Comment (markusahlstrand) [inline src/server.ts:31]: This is no longer needed.

File: src/types/AuthParams.ts [hunk 5]
```diff
@@ -1,7 +1,6 @@
 export enum AuthorizationResponseType {
   TOKEN = "token",
   TOKEN_ID_TOKEN = "token id_token",
-  IMPLICIT = "implicit",
```
Comment (markusahlstrand) [inline src/types/AuthParams.ts:4]: Implicit is not a valid value.

Comment (danielgent) [reply src/types/AuthParams.ts:4]: is `id_token token` though?

On some of the code I've done I've just checked the string contains `id_token`, but it's simpler if it always has to be `token id_token`

File: src/services/token-factory.ts [hunk 6]
```diff
@@ -25,115 +23,124 @@ interface AcessTokenPayload {
...
-        picture:
-          "https://lh3.googleusercontent.com/a/AGNmyxahgsZ1mBDfjYbydgtNDWgS78AvYk68SSoF1it-=s96-c",
-        locale: "en",
-        sid: "BVhF_cwjxxUN5dKFU9b9dc9N9-VyPuLf",
-        updated_at: "2023-04-11T20:33:16.226Z",
-        email,
-      };
+      picture,
+      locale,
+      sid,
```
Comment (danielgent) [inline src/services/token-factory.ts:141]: all looks very familiar but I'm not sure what other option we have :smile: 

Both auth2 and the token service need to support PKCE and access & id tokens

**Pull request merged**